contracts/contracts/paymasters/TimeBasedPaymaster.sol

```diff
@@ -0,0 +1,79 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import {IPaymaster, ExecutionResult, PAYMASTER_VALIDATION_SUCCESS_MAGIC} from "@matterlabs/zksync-contracts/l2/system-contracts/interfaces/IPaymaster.sol";
+import {IPaymasterFlow} from "@matterlabs/zksync-contracts/l2/system-contracts/interfaces/IPaymasterFlow.sol";
+import {TransactionHelper, Transaction} from "@matterlabs/zksync-contracts/l2/system-contracts/libraries/TransactionHelper.sol";
+
+import "@matterlabs/zksync-contracts/l2/system-contracts/Constants.sol";
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+/// @author Matter Labs
+/// @notice This contract validates based on the time of the transaction. 
+contract TimeBasedPaymaster is IPaymaster, Ownable {
+  modifier onlyBootloader() {
+    require(
+      msg.sender == BOOTLOADER_FORMAL_ADDRESS,
+      "Only bootloader can call this method"
+    );
+    _;
+  }
+
+  function validateAndPayForPaymasterTransaction(
+    bytes32,
+    bytes32,
+    Transaction calldata _transaction
+  )
+    external
+    payable
+    onlyBootloader
+    returns (bytes4 magic, bytes memory context)
+  {
+    magic = PAYMASTER_VALIDATION_SUCCESS_MAGIC;
+    require(
+      _transaction.paymasterInput.length >= 4,
+      "The standard paymaster input must be at least 4 bytes long"
+    );
+
+    bytes4 paymasterInputSelector = bytes4(_transaction.paymasterInput[0:4]);
+    if (paymasterInputSelector == IPaymasterFlow.general.selector) {
+      uint256 startTime = (block.timestamp / 86400) * 86400 + 14 hours;
+      uint256 endTime = startTime + 10 minutes;
+
+      require(
+        block.timestamp >= startTime && block.timestamp <= endTime,
+        "Transactions can only be processed between 9:00 AM - 9:03 AM EST"
```

Comment:
Should this time gap not be 10 minutes considering 10min is being added to the start time variable above?

Reply:
Also the times might be incorrect too, I believe it is 14:00 to 14:10?

Reply:
The time is correct, 14:00 is UTC time. I will update nonetheless to be more universal.